specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-05-01-preview/FileImports.json

```diff
@@ -0,0 +1,560 @@
+          "enum": [
+            "CSV",
+            "JSON",
+            "Unspecified"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "fileFormat",
+            "values": [
+              {
+                "description": "A CSV file.",
+                "value": "CSV"
+              },
+              {
+                "description": "A JSON file.",
+                "value": "JSON"
+              },
+              {
+                "description": "A file of other format.",
+                "value": "Unspecified"
+              }
+            ]
+          }
+        },
+        "fileName": {
+          "description": "The name of the file.",
+          "type": "string"
+        },
+        "fileSize": {
+          "description": "The size of the file.",
+          "format": "int32",
+          "type": "integer"
+        },
+        "fileContentUri": {
+          "description": "A URI with a valid SAS token to allow uploading / downloading the file.",
+          "type": "string"
+        },
+        "deleteStatus": {
+          "description": "Indicates whether the file was deleted from the storage account.",
+          "enum": [
+            "Deleted",
+            "NotDeleted",
+            "Unspecified"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "DeleteStatus",
+            "values": [
+              {
+                "description": "The file was deleted.",
+                "value": "Deleted"
+              },
+              {
+                "description": "The file was not deleted.",
+                "value": "NotDeleted"
+              },
+              {
+                "description": "Unspecified",
+                "value": "Unspecified"
+              }
+            ]
+          }
+        }
+      },
+      "type": "object"
+    },
+    "FileImport": {
+      "allOf": [
+        {
+          "$ref": "../../../common/1.0/types.json#/definitions/Resource"
+        }
+      ],
+      "description": "Represents a file import in Azure Security Insights.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/FileImportProperties",
+          "description": "File import properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object"
+    },
+    "FileImportList": {
+      "description": "List all the file imports.",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of file imports.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of file imports.",
+          "items": {
+            "$ref": "#/definitions/FileImport"
+          },
+          "type": "array"
+        }
+      },
+      "type": "object",
+      "required": [
+        "value"
+      ]
+    },
+    "FileImportProperties": {
```

Comment:
>FileImportProperties [](http://example.com/codeflow?start=5&length=20)

Seems like all these properties are set by your RP, mark all properties as readonly: true, also, no need to specify required array since customers won't be setting any of these values.

Reply:
not all properties are set by the RP. for example "FileImportProperties" is an input of the PUT call.